From f5ab8976d9ae4ecacbde8b932b109c1125f8d2b4 Mon Sep 17 00:00:00 2001
From: hwang7 <huanhuan.wang@windriver.com>
Date: Thu, 17 Oct 2013 16:37:07 +0800
Subject: [PATCH] dpa/ethtool: fix oops when use ethtool to get NIC statistics

When using "ethtool -S eth0" to get NIC statistics on t4240,
the kernel  oops.

The reason for this is that the mac_dev's get_stats field is
not initialized.

Signed-off-by: hwang7 <huanhuan.wang@windriver.com>
---
 drivers/net/ethernet/freescale/dpa/mac-api.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/drivers/net/ethernet/freescale/dpa/mac-api.c b/drivers/net/ethernet/freescale/dpa/mac-api.c
index 957c7fa..972e403 100644
--- a/drivers/net/ethernet/freescale/dpa/mac-api.c
+++ b/drivers/net/ethernet/freescale/dpa/mac-api.c
@@ -883,6 +883,7 @@ static void __devinit __cold setup_xgmac(struct mac_device *mac_dev)
 	mac_dev->uninit		= uninit;
 	mac_dev->set_tx_pause	= set_tx_pause;
 	mac_dev->set_rx_pause	= set_rx_pause;
+	mac_dev->get_stats	= mac_get_statistics;
 }
 
 static void __devinit __cold setup_memac(struct mac_device *mac_dev)
@@ -906,6 +907,7 @@ static void __devinit __cold setup_memac(struct mac_device *mac_dev)
 	mac_dev->fm_rtc_set_drift	= fm_rtc_set_drift;
 	mac_dev->fm_rtc_set_alarm	= fm_rtc_set_alarm;
 	mac_dev->fm_rtc_set_fiper	= fm_rtc_set_fiper;
+	mac_dev->get_stats		= mac_get_statistics;
 }
 
 void (*const mac_setup[])(struct mac_device *mac_dev) __devinitconst = {
-- 
1.7.5.4

